pkg/registry/event/registry.go

```diff
@@ -47,6 +47,10 @@ func NewEtcdRegistry(h tools.EtcdHelper, ttl uint64) generic.Registry {
 			TTLFunc: func(runtime.Object, bool) (uint64, error) {
 				return ttl, nil
 			},
+			DeleteCollectionFunc: func(ctx api.Context) (runtime.Object, error) {
+				// TODO Create and return status object
+				return nil, h.Delete(etcdgeneric.NamespaceKeyRootFunc(ctx, prefix), true)
```

Comment:
In that common function, please make sure you return an error if there is no namespace in the provided context to ensure that we do not get into a situation where all resources across all namespaces are inadvertently deleted.